src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -3777,6 +4053,12 @@ arch_emit_simd_intrinsics (const char *class_ns, const char *class_name, MonoCom
 		if (!strcmp (class_name, "Vector128`1") || !strcmp (class_name, "Vector64`1"))
 			return emit_vector64_vector128_t (cfg, cmethod, fsig, args);
 	}
+
+	if (!strcmp (class_ns, "System.Numerics")) {
+		//if (!strcmp ("Vector2", class_name) || !strcmp ("Vector4", class_name) || !strcmp ("Vector3", class_name))
```

Comment:
How could we get Vector2/3 accelerated as well? CoreCLR doesn't handle the underlying types as SIMD types (not 16 bytes aligned and use 8 bytes (Vector2) or 12 byte (Vector3) storage), but instead loads them into SIMD reg using similar to this (example for vector 3):

vmovss      xmm2,dword ptr [r8+8]
vmovsd      xmm3,qword ptr [r8]
vshufps     xmm3,xmm3,xmm2,44h

and when done, storing it back like this:

vmovsd      qword ptr [r8],xmm2
vpshufd     xmm3,xmm2,2
vmovss      dword ptr [r8+8],xmm3

Alternative for Mono is to change Vector2/3 to use new HW intrinsic directly, CoreCLR didn't do the change since they produced worse code compared to regular Vector2/3/4 intrinsics, but if we are not going to accelerate them in MonoVM, then doing HW intrinsics in managed code might at least give us some improvements.

Reply:
It would be nice to use a managed implementation at least on mono, llvm could probably get rid of most of the overhead.

Reply:
I can try something out and see how well it perform compare to CoreCLR's default vector intrinsic on Mono.